board/power_saving.h

```diff
@@ -1,57 +1,45 @@
-#define POWER_SAVE_STATUS_DISABLED 0
-#define POWER_SAVE_STATUS_ENABLED 1
+#include <stdbool.h>
 
-int power_save_status = POWER_SAVE_STATUS_DISABLED;
+extern bool power_save_enabled;
 
-void power_save_enable(void) {
-  if (power_save_status == POWER_SAVE_STATUS_ENABLED) return;
-  puts("enable power savings\n");
+void switch_power_mode(bool enable) {
+  int set_state = enable;
 
-  // turn off can
-  set_can_enable(CAN1, 0);
-  set_can_enable(CAN2, 0);
-  set_can_enable(CAN3, 0);
+  // toggle CAN
+  set_can_enable(CAN1, set_state);
+  set_can_enable(CAN2, set_state);
+  set_can_enable(CAN3, set_state);
 
-  // turn off GMLAN
-  set_gpio_output(GPIOB, 14, 0);
-  set_gpio_output(GPIOB, 15, 0);
+  // toggle GMLAN
+  set_gpio_output(GPIOB, 14, set_state);
+  set_gpio_output(GPIOB, 15, set_state);
 
-  // turn off LIN
-  set_gpio_output(GPIOB, 7, 0);
-  set_gpio_output(GPIOA, 14, 0);
+  // toggle LIN
+  set_gpio_output(GPIOB, 7, set_state);
+  set_gpio_output(GPIOA, 14, set_state);
 
-  if (is_grey_panda) {
-    char UBLOX_SLEEP_MSG[] = "\xb5\x62\x06\x04\x04\x00\x01\x00\x08\x00\x17\x78";
+  if (is_grey_panda == true) {
+    // wake
+    char* UBLOX_POWER_STATE_MSG = "\xb5\x62\x06\x04\x04\x00\x01\x00\x09\x00\x18\x7a";
+    if (enable) {
+      // sleep
+      UBLOX_POWER_STATE_MSG = "\xb5\x62\x06\x04\x04\x00\x01\x00\x08\x00\x17\x78";
+    }
     uart_ring *ur = get_ring_by_number(1);
-    for (int i = 0; i < sizeof(UBLOX_SLEEP_MSG)-1; i++) while (!putc(ur, UBLOX_SLEEP_MSG[i]));
+    for (int i = 0; i < (int)(sizeof(UBLOX_POWER_STATE_MSG))-1; i++) while (!putc(ur, UBLOX_POWER_STATE_MSG[i]));
```

Comment:
This is broken. sizeof(char *) != sizeof (char [])

Reply:
ah right. would we be ok to overwrite the char[] with memcpy, since messages are equal length? or should I rethink this segment?